src/utils/generator.js

```diff
@@ -1367,6 +1367,11 @@ export function attemptToMergeEquivalentObjectAssigns(
       let otherArgsToUse = [];
       for (let x = 1; x < otherArgs.length; x++) {
         let arg = otherArgs[x];
+        // If this arg is a known arg of the root, then we can't merge these Object.assigns
+        // as they share the same args.
+        if (args.includes(arg)) {
```

Comment:
I actually first opted to add a Set and do `has` lookups, but found that it was actually much slower here because the arrays in `Object.assign` are almost always small arrays with generally less than 8 items.